src/component/toolbox/feature/DataView.js

```diff
@@ -93,7 +93,10 @@ function assembleSeriesWithCategoryAxis(series) {
         }));
         var columns = [categoryAxis.model.getCategories()];
         zrUtil.each(group.series, function (series) {
-            columns.push(series.getRawData().mapArray(valueAxisDim, function (val) {
+            // Using `series.name` as dimension when dataset is being used.
+            columns.push(series.getRawData().mapArray(series.dependentModels.dataset.length
+                ? series.name
+                : valueAxisDim, function (val) {
```

Comment:
`series.name` happens to be the dimension name in this case but does not necessarily always be dimension name.
And I think the indent (`, function (val) { ... `) is not good.

I think a simple fix of the issue could be:
```js
var rawData = series.getRawData();
columns.push(rawData.mapArray(rawData.mapDimension(valueAxisDim), function (val) {
    return val;
}));
```